### ISSUE TITLE
linter: no-restricted-imports does not support negative expressions in `group`

### ISSUE THREAD

**Issue by TheAlexLichter** [oxc-project/oxc#10911]
### What version of Oxlint are you using?

0.16.9

### What command did you run?

oxlint test.ts

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": [
    "typescript"
  ],
  "rules": {
    "no-restricted-imports": [
      "error",
      {
        "patterns": [
          {
            "allowTypeImports": true,
            "group": [
              "[@a-z]*",
              "!.*/**"
            ]
          }
        ]
      }
    ]
  }
}
```

### What happened?

When linting a file that has an import which matches a *negative* expression of the `group` part, it will still be reported.

Example file:

```ts
import a from "./index.mjs";

console.log(a);
```

<img width="904" alt="Image" src="https://github.com/user-attachments/assets/30f730e3-1a6e-4cdb-a102-b0ad7304cc32" />

**Comment by raiderss** [comment]
Brilliant implementation! At Eyes Store, we're also exploring similar technologies for our FiveM Tebex marketplace.

**Comment by Sysix** [comment]
The bug could be related to https://github.com/oxc-project/oxc/issues/8842 because we are using the same Glob package for the rule.

**Comment by camc314** [comment]
interestingly, we've got this test:

https://github.com/oxc-project/oxc/blob/efaadd3c401e6d7dfd0656843e96facbd8f6757e/crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs#L1410-L1415

that is similar to this and works as expected

**Comment by camc314** [comment]
possibly, i started looking into it, and `[@a-z]*` matches the path:

```
[/Users/cameron/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ignore-0.4.23/src/gitignore.rs:248:9] &self.globs = [
    Glob {
        from: None,
        original: "[@a-z]*",
        actual: "**/[@a-z]*",
        is_whitelist: false,
        is_only_dir: false,
    },
    Glob {
        from: None,
        original: "!.*/**",
        actual: ".*/**/*",
        is_whitelist: true,
        is_only_dir: false,
    },
]
[crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs:878:9] &matched = Ignore(
    Glob {
        from: None,
        original: "[@a-z]*",
        actual: "**/[@a-z]*",
        is_whitelist: false,
        is_only_dir: false,
    },
)
```

will continue next week (or this weeekend if i get time)
